notification/src/handler/notification/notification.handler.js

```diff
@@ -6,6 +6,31 @@ const { getNotificationForTracking } = require('../../utils/commons')
 const ctp = require('../../utils/ctp')
 const mainLogger = require('../../utils/logger').getLogger()
 
+class CommercetoolsError extends Error {
+  constructor({ stack, message, statusCode }) {
+    super()
+    this.stack = stack
+    this.message = message
+    this.isRecoverable = this._isRecoverable(statusCode)
+  }
+
+  /**
+   * recoverable: notification delivery can be retried by Adyen
+   * non recoverable: notification delivery can not be retried by Adyen as it most probably would fail again
+   * If commercetools status code is defined and is 5xx then return `500` to Adyen -> recoverable
+   * If during communication with commercetools we got a `NetworkError` then return `500` -> recoverable
+   * If commercetools status code is not OK but also not 5xx or 409 then return `accepted` -> non recoverable
+   */
+  _isRecoverable(statusCode) {
```

Comment:
`isRecoverable` seems confusing to understand. Maybe `shouldRetry` or just `retry` would be better?

Sorry, I did not came with that earlier but some things just need multiple iterations :)

Reply:
5e48180

Reply:
no worries, I am trying to keep changes to a minimum as PR is not focusing on the refactoring much, thanks for the suggestions.